src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -243,6 +280,33 @@ public AzureVMAgentTemplate(
         labelDataSet = Label.parse(labels);
     }
 
+    public static Map<String, Object> getTemplateProperties(AzureVMAgentTemplate template) {
+        Map<String, Object> templateProperties = new HashMap<>();
+        String builtInImage = template.getBuiltInImage();
+        Map<String, String> defaultProperties = AzureVMManagementServiceDelegate.DEFAULT_IMAGE_PROPERTIES.get(builtInImage);
+        Boolean isBasic = template.isTopLevelType("basic");
+
+        templateProperties.put("imagePublisher", isBasic ? defaultProperties.get(Constants.DEFAULT_IMAGE_PUBLISHER) : template.getImagePublisher());
+        templateProperties.put("imageOffer", isBasic ? defaultProperties.get(Constants.DEFAULT_IMAGE_OFFER) : template.getImageOffer());
+        templateProperties.put("imageSku", isBasic ? defaultProperties.get(Constants.DEFAULT_IMAGE_SKU) : template.getImageSku());
+        templateProperties.put("imageVersion", isBasic ? defaultProperties.get(Constants.DEFAULT_IMAGE_VERSION) : template.getImageVersion());
+        templateProperties.put("osType", isBasic ? defaultProperties.get(Constants.DEFAULT_OS_TYPE) : template.getOsType());
+        templateProperties.put("agentLaunchMethod", isBasic ? defaultProperties.get(Constants.DEFAULT_LAUNCH_METHOD) : template.getAgentLaunchMethod());
+        templateProperties.put("initScript", isBasic ? AzureVMManagementServiceDelegate.DEFAULT_INIT_SCRIPT.get(builtInImage) : template.getInitScript());
+        templateProperties.put("virtualNetworkName", isBasic ? "" : template.getVirtualNetworkName());
```

Comment:
No, I think these advanced configurations are too complex that ordinary users won't change them.